### PR TITLE
Location.get_rise_set_transit

### LINKED ISSUE #114: add sunrise / sunset to Location
In reference to changes in #93:

add the following functions to `:py:class:`Location``:

`get_sunrise` and `get_sunset` and deferred from these `get_dayduration`.

This helps filtering values, e.g. set night values to `np.nan`.

### PR DESCRIPTION
**Pull request by cwhanse**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python! You may delete all of these instructions except for the list below.

You may submit a pull request with your code at any stage of completion.

The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items below:

-  [x] Closes #114
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

`Location` method wrapping `sun_rise_set_transit_xxx` functions.

### CODE REVIEW

**Comment by cwhanse** [comment]
Help me understand this test failure: `test_location` is importing a `Location` object `golden` from `test_solarposition` but does not bring in the new `Location` method `get_sun_rise_set_transit`.  Do I need to redeclare `golden` in `test_location`?

**Comment by cwhanse** [comment]
Ready for review

**Comment by cwhanse** [comment]
Test failure is on py2.7-min, because functions being tested require pyephem.

File: pvlib/location.py [hunk 1]
```diff
@@ -276,3 +276,44 @@ def get_airmass(self, times=None, solar_position=None,
...
+        elif method == 'geometric':
+            result = solarposition.sun_rise_set_transit_geometric(times,
+                                                            self.latitude,
+                                                            self.longitude,
+                                                            **kwargs)
+        else:
+            raise ValueError('{} is not a valid method. Must be '
+                             'one of pyephem, spa, geometric'
+                             .format(method))
+        return result
```
Comment (mikofski) [inline pvlib/location.py]: I was thinking that here would be the place that you would convert the result to a dataframe, if a dataframe was passed in. IE: the low level functions return low level return values, while the high level function is more polite.

It makes sense to me to group functions in this hierarchy of low-level vs. high-level, because then the task of converting the result to a dataframe is DRY, it only occur once, rather than repeating WET code in multiple locations.

Comment (cwhanse) [reply pvlib/location.py]: That's a good catch. Changed to dataframe output, thanks!

**Comment by cwhanse** [comment]
`pvlib.SPA` error when testing in py36

File: pvlib/location.py (hunk 1 above)
Comment (wholmgren) [reply pvlib/location.py]: Hard to follow the history here. Is this comment still relevant or can it be marked as resolved? Github does not yet show it as outdated. It looks ok as of dd8d621

**Comment by wholmgren** [comment]
> pvlib.SPA error when testing in py36

Strange. The base python is 3.6 but tests run in a conda env with 3.7. I suspect it's related to conda packaging issues that will resolve themselves. That configuration passed on master 8 days ago, but failed on master when I reran it a few minutes ago. So that confirms this PR is not doing something unexpected.

File: pvlib/location.py (hunk 1 above)
Comment (cwhanse) [reply pvlib/location.py]: Resolved.

**Comment by wholmgren** [comment]
@cwhanse the test failure is unrelated. You can merge this if you're ready.
